kernelci/runtime/kubernetes.py

```diff
@@ -22,18 +22,19 @@ class Kubernetes(Runtime):
 
     JOB_NAME_CHARACTERS = string.ascii_lowercase + string.digits
 
-    def generate(self, params, plan_config):
-        template = self._get_template(plan_config)
+    @classmethod
+    def _get_job_file_name(cls, params):
+        return '.'.join([params['k8s_job_name'], 'yaml'])
```

Comment:
isn't it the same as `params['k8s_job_name'] + '.yaml'`?

Reply:
I've always thought that join was better and faster than + for joining strings so I'm always using this approach. I think there's an article somewhere comparing these things but I read it several years ago.

Reply:
Also it's not a change related to this PR either, it's only reworking the method as a private class one.

Reply:
> I've always thought that join was better and faster than + for joining strings so I'm always using this approach. I think there's an article somewhere comparing these things but I read it several years ago.

Yes, in general it is, I think still the fastest method of string concatenation and + is the slowest. I put '+' expression there as an example of what I have in mind. Maybe not clearly enough. `.join` is the fastest, slower, but still fast is `f-string`, the slowest is `+`. However, in this case we're joining 2 strings and the speed is not the priority, but "readability counts` principle is still in use. This is just for the sake of clarification.

If it's not part of your PR, feel free to ignore it. I'll possible convert it to an f-string at some other occasion.

Reply:
Right, well once we have the whole pipeline actually functional then we can review the whole code and tweak tiny details like this I guess.